src/cargo/ops/registry.rs

```diff
@@ -560,6 +560,9 @@ pub fn configure_http_handle(config: &Config, handle: &mut Easy) -> CargoResult<
     if let Some(proxy) = http_proxy(config)? {
         handle.proxy(&proxy)?;
     }
+    handle.proxy_auth(&http.proxy_auth.to_easy())?;
```

Comment:
If this is `CargoHttpProxyAuth::Disable` should we avoid calling any of the `proxy_` methods?